flytekit/common/tasks/raw_container.py

```diff
@@ -0,0 +1,255 @@
+            image: str,
+            outputs: Dict[str, FlyteSdkType]=None,
+            input_data_dir: str = None,
+            output_data_dir: str = None,
+            metadata_format: str = METADATA_FORMAT_JSON,
+            command: List[str] = None,
+            args: List[str] = None,
+            storage_request: str = None,
+            cpu_request: str = None,
+            gpu_request: str = None,
+            memory_request: str = None,
+            storage_limit: str = None,
+            cpu_limit: str = None,
+            gpu_limit: str = None,
+            memory_limit: str = None,
+            environment: Dict[str, str] = None,
+            interruptible: bool = False,
+            discoverable: bool = False,
+            discovery_version: str = None,
+            retries: int = 1,
+            timeout: _datetime.timedelta = None,
+    ):
+        """
+        :param inputs:
+        :param outputs:
+        :param image:
+        :param command:
+        :param args:
+        :param storage_request:
+        :param cpu_request:
+        :param gpu_request:
+        :param memory_request:
+        :param storage_limit:
+        :param cpu_limit:
+        :param gpu_limit:
+        :param memory_limit:
+        :param environment:
+        :param interruptible:
+        :param discoverable:
+        :param discovery_version:
+        :param retries:
+        :param timeout:
+        :param input_data_dir: This is the directory where data will be downloaded to
+        :param output_data_dir: This is the directory where data will be uploaded from
+        :param metadata_format: Format in which the metadata will be available for the script
+        """
+
+        # Set as class fields which are used down below to configure implicit
+        # parameters
+        self.input_data_dir = input_data_dir
+        self.output_data_dir = output_data_dir
+        self.metadata_format = metadata_format
+        if metadata_format not in self._METADATA_FORMAT:
+            raise Exception("{} Illegal metadata format, only [{}] metadata formats are supported".format(metadata_format, ",".join(self._METADATA_FORMAT)))
+
+        metadata = _task_models.TaskMetadata(
+            discoverable,
+            # This needs to have the proper version reflected in it
+            _task_models.RuntimeMetadata(
+                _task_models.RuntimeMetadata.RuntimeType.FLYTE_SDK, __version__,
+                "python"),
+            timeout or _datetime.timedelta(seconds=0),
+            _literals.RetryStrategy(retries),
+            interruptible,
+            discovery_version,
+            "This is deprecated!"
+        )
+
+        # Here we set the routing_group, catalog, and schema as implicit
+        # parameters for caching purposes
+        i = _interface.TypedInterface(inputs=types_to_variable(inputs), outputs=types_to_variable(outputs))
+
+        # TODO create custom proto to store data dir and other things
```

Comment:
remove

Reply:
no not remove, add the custom proto I guess?